src/Language/PureScript/AST/Traversals.hs

```diff
@@ -53,7 +53,8 @@ everywhereOnValues f g h = (f', g', h')
   where
   f' :: Declaration -> Declaration
   f' (DataBindingGroupDeclaration ds) = f (DataBindingGroupDeclaration (fmap f' ds))
-  f' (ValueDeclaration sa name nameKind bs val) = f (ValueDeclaration sa name nameKind (fmap h' bs) (fmap (mapGuardedExpr handleGuard g') val))
+  f' (ValueDeclaration (ValueDeclarationData sa name nameKind bs val)) =
```

Comment:
Maybe I'm not seeing it, but 

`f (ValueDeclaration (map (mapGuardedExpr handleGuard g') <$> ValueDeclarationData sa name nameKind (fmap h' bs) val))`

doesn't seem better than:

`f (ValueDeclaration (ValueDeclarationData sa name nameKind (fmap h' bs) (fmap (mapGuardedExpr handleGuard g') val)))`

What's the win? We still need to destructure to get at the binders.